third_party/blink/renderer/core/html/media/html_media_element.cc

```diff
@@ -405,26 +405,51 @@ std::ostream& operator<<(std::ostream& stream,
   return stream << static_cast<void const*>(&media_element);
 }
 
+#if BUILDFLAG(USE_STARBOARD_MEDIA)
+// Checks for progressive formats served by the YouTube H5 player.
+// These formats have a mime type of "video/mp4", and lists both audio and
+// video formats under the "codecs" parameter.
+bool IsProgressiveFormat(const ContentType& content_type) {
+  String type = content_type.GetType();
+  String codecs = content_type.Parameter("codecs");
+  if (!type.empty() && !codecs.empty()) {
+    Vector<String> split_codecs;
+    String separator(", ");
+    codecs.Split(separator, split_codecs);
+    return type.Utf8() == "video/mp4" && split_codecs.size() == 2;
+  }
+  return false;
+}
+#endif  // BUILDFLAG(USE_STARBOARD_MEDIA)
+
 }  // anonymous namespace
 
 // static
 MIMETypeRegistry::SupportsType HTMLMediaElement::GetSupportsType(
     const ContentType& content_type) {
 #if BUILDFLAG(USE_STARBOARD_MEDIA)
   // Interupt Chromium's IsTypeSupported() from here for better performance.
-  SbMediaSupportType support_type =
-      SbMediaCanPlayMimeAndKeySystem(content_type.Raw().Ascii().c_str(), "");
   MIMETypeRegistry::SupportsType result;
-  switch (support_type) {
-    case kSbMediaSupportTypeNotSupported:
-      result = MIMETypeRegistry::kNotSupported;
-      break;
-    case kSbMediaSupportTypeMaybe:
-      result = MIMETypeRegistry::kMaybeSupported;
-      break;
-    case kSbMediaSupportTypeProbably:
-      result = MIMETypeRegistry::kSupported;
-      break;
+  if (!base::FeatureList::IsEnabled(media::kCobaltProgressivePlayback) &&
+      IsProgressiveFormat(content_type)) {
+    LOG(INFO) << "Content type \'" << content_type.Raw()
```

Comment:
If we return `DEMUXER_ERROR_COULD_NOT_OPEN` in demuxer_manager.cc, do we still need this? That is, if returning DEMUXER_ERROR_COULD_NOT_OPEN is sufficient for informing H5 Player, then we can skip this part.

Reply:
We likely don't, however I don't know if this code path is triggered if the src is set without first calling `CanPlayType()`. It may also be more desirable to reject here instead of in the demuxer manager after playback is initiated, but we still have the check in demuxer manager as a fallback.

Reply:
My understanding is that if we disable ffmpeg_demuxer and progressive_demuxer, we don't have any fallback for this, right? in this case, even we reject the format, Cobalt still cannot play it.

Reply:
Ah, I meant fallback as a way to prevent the ProgressiveDemuxer from being constructed if we don't reject support in CanPlayType(). We wouldn't play the content either way

Reply:
So we probably don't need to reject this format in advance, as [WebMediaPlayerImpl](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/platform/media/web_media_player_impl.cc;l=3039;drc=616d60fca655937c2b730db94fd32d37ddff3bb5;bpv=0;bpt=1) will report an error when it failed to create the demuxer.

We just need to ensure to send the log message to H5 Player, so we see some numbers of playback errors, it is due to this failing construction.

Reply:
I get what you're saying. Xiaoming put in the initial request to check in both locations, I've asked him if he feels strongly one way or the other.

Reply:
Sure, I'll leave both in and let Px know.